File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
import java.util.*;
class Solution {
    public int[] solution(int n, int s) {
        int[] ans = new int[n];
        if (n > s) {return new int[] {-1};}
        int q = s / n;
        int r = s % n;
        for (int i = 0; i < n-r; i++) {
            ans[i] = q;
        }
        for (int i = n-r; i < n; i++) {
            ans[i] = q+1;
        }
        return ans;
    }
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    ans = []
    # n개의 원소 합이 s면서 곱 최대 => ele 간의 격차 최소화
    # s를 n으로 나누고 부족한 건 n-1에 나눠주기
    if s < n: return [-1]
    q,r = divmod(s,n)
    arr = [q]*n
    for i in range(r):
        arr[i] += 1
    return sorted(arr) if arr else [-1]